tests/drivers/get_binding/pinmux/testcase.yaml

```diff
@@ -0,0 +1,4 @@
+tests:
+  test_get_binding:
+    tags: drivers
+    depends_on: pinmux
```

Comment:
if the intention to test this with the native_posix port, it will not run...
We probably need to use a dummy driver here for testing purposes

Reply:
I was thinking the same, we can add a test driver of misc type and use it for all device api testing.

@lpereira can you move this test out of pimux folder to test/drivers/get_binding/ and in prj.conf we can specify the name we want to use. For now we can assign CONFIF_PINMUX_NAME, This way we ware not binding the test to pinmux, later we can change to the appropriate one.

Reply:
i am playing with such a test right now, will post for review in a bit.

Reply:
:)

Reply:
see #6213